packages/explorer/src/common.ts

```diff
@@ -0,0 +1,9 @@
+import { Chain, anvil, garnet, redstone } from "viem/chains";
+
+const chains = {
+  [anvil.id]: anvil,
+  [redstone.id]: redstone,
+  [garnet.id]: garnet,
+};
+export const chainId = Number(process.env.NEXT_PUBLIC_CHAIN_ID || anvil.id);
+export const chain = chains[chainId as keyof typeof chains] as Chain;
```

Comment:
might be worth checking+throwing if `chainId` isn't in `chains`?

Reply:
TS can guide you to this if you change the type to `Partial<Record<number, Chain>>`

Reply:
nice tip!